src/Drupal/Commands/core/FieldDeleteCommands.php

```diff
@@ -0,0 +1,150 @@
+                '%optionName' => 'field-name',
+            ]));
+        }
+
+        /** @var FieldConfig[] $results */
+        $results = $this->entityTypeManager
+            ->getStorage('field_config')
+            ->loadByProperties([
+                'field_name' => $fieldName,
+                'entity_type' => $entityType,
+                'bundle' => $bundle,
+            ]);
+
+        if ($results === []) {
+            throw new \InvalidArgumentException(
+                t("Field with name ':fieldName' does not exist on bundle ':bundle'.", [
+                    ':fieldName' => $fieldName,
+                    ':bundle' => $bundle,
+                ])
+            );
+        }
+
+        $this->deleteFieldConfig(reset($results));
+
+        // Fields are purged on cron. However field module prevents disabling modules
+        // when field types they provided are used in a field until it is fully
+        // purged. In the case that a field has minimal or no content, a single call
+        // to field_purge_batch() will remove it from the system. Call this with a
+        // low batch limit to avoid administrators having to wait for cron runs when
+        // removing fields that meet this criteria.
+        field_purge_batch(10);
+    }
+
+    protected function askExisting(string $entityType, string $bundle): string
+    {
+        $choices = [];
+        /** @var FieldConfigInterface[] $fieldConfigs */
+        $fieldConfigs = $this->entityTypeManager
+            ->getStorage('field_config')
+            ->loadByProperties([
+                'entity_type' => $entityType,
+                'bundle' => $bundle,
+            ]);
+
+        foreach ($fieldConfigs as $fieldConfig) {
+            $label = $this->input->getOption('show-machine-names')
+                ? $fieldConfig->get('field_name')
+                : $fieldConfig->get('label');
+
+            $choices[$fieldConfig->get('field_name')] = $label;
+        }
+
+        return $this->io()->choice('Choose a field to delete', $choices);
+    }
+
+    protected function deleteFieldConfig(FieldConfigInterface $fieldConfig): void
+    {
+        $fieldStorage = $fieldConfig->getFieldStorageDefinition();
+        $bundles = $this->entityTypeBundleInfo->getBundleInfo($fieldConfig->getTargetEntityTypeId());
+        $bundleLabel = $bundles[$fieldConfig->getTargetBundle()]['label'];
+
+        if ($fieldStorage && !$fieldStorage->isLocked()) {
+            $fieldConfig->delete();
+
+            // If there is only one bundle left for this field storage, it will be
+            // deleted too, notify the user about dependencies.
+            if (count($fieldStorage->getBundles()) <= 1) {
+                $fieldStorage->delete();
+            }
+
+            $message = 'The field :field has been deleted from the :type content type.';
+        } else {
+            $message = 'There was a problem removing the :field from the :type content type.';
```

Comment:
I used the exact same error message as shown in the UI: https://git.drupalcode.org/project/drupal/-/blob/9.3.x/core/modules/field_ui/src/Form/FieldConfigDeleteForm.php#L104

Isn't it possible that the field storage being locked can have other causes? Because if that's the case, we should keep the error message generic, just like `field_ui` does. Btw, I just tested it and I can delete fields with data in them using `field:delete`.